src/config-utils.ts

```
@@ -468,6 +471,13 @@ async function getLanguages(): Promise<string[]> {
     core.info("Automatically detected languages: " + JSON.stringify(languages));
   }
 
+  // If the languages parameter was not given and no languages were
+  // detected then fail here as this is a workflow configuration error.
+  if (languages.length === 0) {
+    throw new Error("Did not detect any languages to analyze. " +
+        "Please update input in workflow or check that GitHub detects the correct languages in your repository.");
+  }
+
```

Sorry if this is a silly question @robertbrignull but will the error appear more prominently (in the Check failure summary, rather than only in the detailed logs) if thrown from here rather than in the calling method as it was before these changes? Or do we also need to change _how_ we throw the error?

---

Where the error is thrown from doesn't make a difference to how it is presented.

I'm going to do a test of this to make sure it works and looks presentable before this is merged.

---

:+1: I'd love a link to see what it looks like too!